main.py

```diff
@@ -142,28 +174,29 @@ async def get_experiment_info(file: str) -> Any:
     return remote.examine(file)
 
 
-previous_queue = {}
-
+@app.get("/experiment/queue/", response_model=ScheduleInfo)
+async def get_scheduled_queue(updated_time: Optional[float] = None) -> Any:
+    """Gets the scheduled queue and returns it.
 
-@app.get("/experiment/queue/")
-async def get_experiment_queue() -> Dict[int, Dict[str, Any]]:
-    """Gets the list of queued experiment and returns it.
+    Args:
+        updated_time: The last updated time by the client, in the format of time.time().
 
     Returns:
-        A dictionary of queued experiments with rid as their keys.
-        The value of each corresponding rid is a dictionary with several information:
-          "priority", "status", "due_date", "pipeline", "expid", etc.
-        It includes the running experiment with different "status" value.
+        The latest schedule info.
+        If the given updated time is earlier than when the latest schedule was updated,
+        it returns the latest schedule info immediately.
+        Otherwise, it polls until the schedule is updated and then returns it.
     """
+    if updated_time is None or updated_time < latest_schedule.updated_time:
+        return latest_schedule
     remote = get_client("master_schedule")
-    current_queue = copy.deepcopy(previous_queue)
-    while current_queue == previous_queue:
+    latest_queue = copy.deepcopy(latest_schedule.queue)
+    queue = copy.deepcopy(latest_queue)
```

Comment:
How about `queue = remote.get_status()`?

Reply:
Wow, it looks much better👍